File: project/src/main/java/com/example/project/entities/TimeActive.java
package com.example.project.entities;

import java.io.Serializable;
import java.util.Date;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.Data;


@Data
@Document(collection = "TimeActive")
public class TimeActive implements Serializable {
	@Id
	private int timeActive_id;
	private Date start_time;
	private Date end_time;
	private String status;
	private int total_time;
    private Sensor sensor;
	
}

File: project/angular_project/src/app/models/time-active.ts
import { Sensor } from "./sensor";

export class TimeActive {
  timeActive_id: number;
  start_time: Date;
  end_time: Date;
  status: string;
  total_time: number;
  sensor: Sensor;

  constructor() {
    this.timeActive_id = 0;
    this.start_time = new Date();
    this.end_time = new Date();
    this.status = '';
    this.total_time = 0;
    this.sensor = new Sensor();
  }
}
